src/app/shared/services/app.service.ts

```diff
@@ -61,10 +61,48 @@ export class NSSService {
     public setAddScenarioModal(val: any) { 
         this._showHideAddScenarioModal.next(val);
     }
+    // -+-+-+-+-+-+-+-+-+ clone scenario modal -+-+-+-+-+-+-+-+
+    private _showHideCloneScenarioModal: Subject<boolean> = new Subject<boolean>();
+    public setCloneScenarioModal(val: any) { 
+        this._showHideCloneScenarioModal.next(val);
+    }
+
+    private statisticGroupSource = new BehaviorSubject<any>(' ');
+    currentStatisticGroup=this.statisticGroupSource.asObservable();
+
+    changeStatisticGroup(statisticGroup: any){
+        this.statisticGroupSource.next(statisticGroup);
+    }
+
+    private itemSource = new BehaviorSubject<any>(' ');
+    currentItem=this.itemSource.asObservable();
+
+    changeItem(item: any){
+        this.itemSource.next(item);
+    }
+
+    private regRegionSource = new BehaviorSubject<any>(' ');
+    currentRegRegion=this.regRegionSource.asObservable();
+
+    changeRegRegion(regRegion: any){
+        this.regRegionSource.next(regRegion);
+    }
+
+    private cloneSource = new BehaviorSubject<boolean>(true);
+    currentClone=this.cloneSource.asObservable();
+
+    changeClone(clone: boolean){
+        this.cloneSource.next(clone);
+    }
+
     // show the add scenario modal in the mainview
     public get showAddScenarioModal(): any {
         return this._showHideAddScenarioModal.asObservable();
     }
+    // show the add scenario modal in the mainview
+    public get showCloneScenarioModal(): any {
+        return this._showHideCloneScenarioModal.asObservable();
+    }
```

Comment:
A lot of this will be cleaned up if we decide to just send one object from the mainview when we clone